src/PerfView/StackViewer/StackWindow.xaml.cs

```diff
@@ -3615,44 +3615,32 @@ private bool ValidateStartAndEnd(StackSource newSource)
                     EndTextBox.Text = (newSource.SampleTimeRelativeMSecLimit).ToString("n3");
                 }
             }
-            else if (!double.TryParse(EndTextBox.Text, out val2))
+            else if (!double.TryParse(EndTextBox.Text, out end))
             {
                 StatusBar.LogError("Invalid number " + EndTextBox.Text);
                 EndTextBox.Text = "Infinity";
                 return false;
             }
             else
             {
-                EndTextBox.Text = val2.ToString("n3");
+                EndTextBox.Text = end.ToString("n3");
             }
 
             // See if we are pasting a range into the start text box
-            if (double.TryParse(StartTextBox.Text, out val1))
+            if (double.TryParse(StartTextBox.Text, out start))
             {
-                StartTextBox.Text = val1.ToString("n3");
+                StartTextBox.Text = start.ToString("n3");
             }
             else if (string.IsNullOrWhiteSpace(StartTextBox.Text))
             {
                 StartTextBox.Text = "0";
             }
             else
             {
-                // TODO: This only works for cultures where a space is not the numeric group separator
-                var match = Regex.Match(StartTextBox.Text, @"^\s*([\d\.,]+)\s+([\d\.,]+)\s*$");
-                if (match.Success)
+                if (RangeUtilities.TryParse(StartTextBox.Text, out start, out end))
                 {
-                    if (double.TryParse(match.Groups[1].Value, out val1) &&
-                        double.TryParse(match.Groups[2].Value, out val2))
-                    {
-                        StartTextBox.Text = val1.ToString("n3");
-                        EndTextBox.Text = val2.ToString("n3");
-                    }
-                    else
-                    {
-                        StatusBar.LogError("Invalid number " + StartTextBox.Text);
```

Comment:
We should preserve the behavior where we log the error here if something goes wrong.

Reply:
What logging are we missing?

Reply:
The call to StatusBar.LogError("Invalid number " + StartTextBox.Text);

---
In reply to: [524065367](https://github.com/microsoft/perfview/pull/1288#discussion_r524065367) [](ancestors = 524065367)